internal/usde/usde.go

```diff
@@ -53,51 +35,80 @@ func DeriveUpgradeStrategy(ctx context.Context, newSpec *kubernetes.GenericObjec
 	numaLogger.WithValues("usdeConfig", usdeConfig, "applyPaths", applyPaths).Debug("started deriving upgrade strategy")
 
 	// Split newSpec
-	newSpecOnlyApplyPaths, newSpecWithoutApplyPaths, err := util.SplitObject(newSpec.Spec.Raw, applyPaths, ".")
+	newSpecOnlyApplyPaths, newSpecWithoutApplyPaths, err := util.SplitObject(newSpec.Spec.Raw, applyPaths, []string{}, ".")
 	if err != nil {
-		return UpgradeStrategyError, nil, err
+		return false, apiv1.UpgradeStrategyError, err
 	}
 
-	numaLogger.WithValues("newSpecOnlyApplyPaths", newSpecOnlyApplyPaths, "newSpecWithoutApplyPaths", newSpecWithoutApplyPaths).Debug("split new spec")
+	numaLogger.WithValues(
+		"newSpecOnlyApplyPaths", newSpecOnlyApplyPaths,
+		"newSpecWithoutApplyPaths", newSpecWithoutApplyPaths,
+	).Debug("split new spec")
 
 	// Split existingSpec
-	existingSpecOnlyApplyPaths, existingSpecWithoutApplyPaths, err := util.SplitObject(existingSpec.Spec.Raw, applyPaths, ".")
+	existingSpecOnlyApplyPaths, existingSpecWithoutApplyPaths, err := util.SplitObject(existingSpec.Spec.Raw, applyPaths, []string{}, ".")
 	if err != nil {
-		return UpgradeStrategyError, nil, err
+		return false, apiv1.UpgradeStrategyError, err
 	}
 
-	numaLogger.WithValues("existingSpecOnlyApplyPaths", existingSpecOnlyApplyPaths, "existingSpecWithoutApplyPaths", existingSpecWithoutApplyPaths).Debug("split existing spec")
-
-	specsDiffer := true
+	numaLogger.WithValues(
+		"existingSpecOnlyApplyPaths", existingSpecOnlyApplyPaths,
+		"existingSpecWithoutApplyPaths", existingSpecWithoutApplyPaths,
+	).Debug("split existing spec")
 
 	// Compare specs without the apply fields and check user's strategy to either return PPND or Progressive
 	if !reflect.DeepEqual(newSpecWithoutApplyPaths, existingSpecWithoutApplyPaths) {
-		namespaceConfig := config.GetConfigManagerInstance().GetNamespaceConfig(newSpec.Namespace)
-
-		var userUpgradeStrategy config.USDEUserStrategy = ""
-		if namespaceConfig != nil {
-			userUpgradeStrategy = namespaceConfig.UpgradeStrategy
+		userUpgradeStrategy, err := GetUserStrategy(ctx, newSpec.Namespace)
+		if err != nil {
+			return false, apiv1.UpgradeStrategyError, err
 		}
 
-		numaLogger.WithValues("userUpgradeStrategy", userUpgradeStrategy).Debug("the specs without the 'apply' paths are different")
-
-		if userUpgradeStrategy == config.PPNDStrategyID {
-			return UpgradeStrategyPPND, &specsDiffer, nil
+		numaLogger.WithValues(
+			"userUpgradeStrategy", userUpgradeStrategy,
+			"newSpecWithoutApplyPaths", newSpecWithoutApplyPaths,
+			"existingSpecWithoutApplyPaths", existingSpecWithoutApplyPaths,
+		).Debug("the specs without the 'apply' paths are different")
+
+		switch userUpgradeStrategy {
+		case config.PPNDStrategyID:
+			return true, apiv1.UpgradeStrategyPPND, nil
+		case config.ProgressiveStrategyID:
+			return true, apiv1.UpgradeStrategyProgressive, nil
+		case config.NoStrategyID:
+			return true, apiv1.UpgradeStrategyApply, nil
```

Comment:
This block of code according to the comment at L59 should only return `PPND or Progressive`?

Reply:
Good catch that that's a mismatch. I am going to change the comment. Perhaps in the future, those would be the only strategies allowed, but for now "no strategy" is also allowed.